x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/20_search_application_put.yml

```diff
@@ -109,29 +115,34 @@ teardown:
         body:
           indices: [ "test-index1", "test-index2" ]
           template:
-            script: {
-              source: "{ \"query\": { \"query_string\":{ \"query\": \"{{query_string}}\", \"default_field\": \"_all\" } } }",
-              params: {
+            script:
+              source: "{ \"query\": { \"query_string\":{ \"query\": \"{{query_string}}\", \"default_field\": \"_all\" } } }"
+              params:
                 query_string: "elastic"
-              },
               lang: "mustache"
-            }
+            dictionary:
+              properties:
+                another_query_string:
+                  type: "string"
 
   - do:
       search_application.get:
         name: another-test-search-application
-  - match: {
-    template: {
-      script: {
-        source: "{ \"query\": { \"query_string\":{ \"query\": \"{{query_string}}\", \"default_field\": \"_all\" } } }",
-        params: {
-          query_string: "elastic"
-        },
-        lang: "mustache"
-      }
-    }
-  }
+
+  - match:
+      template:
+        script:
+          source: "{ \"query\": { \"query_string\":{ \"query\": \"{{query_string}}\", \"default_field\": \"_all\" } } }"
+          params:
+            query_string: "elastic"
+          lang: "mustache"
+        dictionary:
+          properties:
+            another_query_string:
```

Comment:
Same question as above: Do we want this request to succeed? Even though it's string encoded, this template is specifying the parameter query_string and it's not in the dictionary.